File: exos_openclassrooms/correction/mini-chat1/mini_chat.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;
namespace New_test
{
    #region Mini_chat
    public class Mini_chat
    {
        #region Member Variables
        protected int _id;
        protected string _nom;
        protected string _message;
        protected DateTime _date;
        #endregion
        #region Constructors
        public Mini_chat() { }
        public Mini_chat(string nom, string message, DateTime date)
        {
            this._nom=nom;
            this._message=message;
            this._date=date;
        }
        #endregion
        #region Public Properties
        public virtual int Id
        {
            get {return _id;}
            set {_id=value;}
        }
        public virtual string Nom
        {
            get {return _nom;}
            set {_nom=value;}
        }
        public virtual string Message
        {
            get {return _message;}
            set {_message=value;}
        }
        public virtual DateTime Date
        {
            get {return _date;}
            set {_date=value;}
        }
        #endregion
    }
    #endregion
}

File: exos_openclassrooms/correction/mini-chat1/minichat.php

<?php session_start(); ?>

<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8" />
        <title>Mini-chat</title>
    </head>
    <style>
    form
    {
        text-align:center;
    }
    </style>
    <body>
        
    <form action="minichat_post.php" method="post">
        <p>

        <label for="pseudo">Pseudo</label> : <input type="text" name="pseudo" id="pseudo" value="<?php echo $_SESSION['session_pseudo']; ?>" /><br />
        <label for="message">Message</label> :  <input type="text" name="message" id="message" /><br />
        <input type="submit" value="Envoyer" />
        <input type="button" onclick="header('Location: minichat.php');" value="Rafraichir">
    </p>
    </form>

<?php
// Connexion à la base de données
try
{
    $bdd = new PDO('mysql:host=localhost;dbname=oc;charset=utf8', 'root', '', array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION));
}
catch(Exception $e)
{
        die('Erreur : '.$e->getMessage());
}

// Récupération des 10 derniers messages
$reponse = $bdd->query('SELECT nom, message, DATE_FORMAT(date, \'le %d/%m/%Y à %Hh %imin %ss\') AS date_fr FROM mini_chat ORDER BY ID DESC LIMIT 0, 10');

// Affichage de chaque message (toutes les données sont protégées par htmlspecialchars)
// Affichage de la date en "italique"
while ($donnees = $reponse->fetch())
{
    echo '<p><strong>' . htmlspecialchars($donnees['nom']) . '</strong> : ' . htmlspecialchars($donnees['message']) . '<br /><font size="2" color="blue">' . htmlspecialchars($donnees['date_fr']).'</p></font>'; 
}

$reponse->closeCursor();

?>
    </body>
</html>